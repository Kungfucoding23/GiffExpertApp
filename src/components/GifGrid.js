import React from 'react'

export const GifGrid = ({category}) => {

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=KJYm7HX3CDWX39cn8b0AJKzwGZFDzKBG';
        const resp = await fetch(url);
        const {data} = await resp.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url  // EL ? ES PARA PREGUNTAR QUE SI VIENEN LAS IMAGENS ENTONCES LAS UTILICE
            }
        });

        console.log(gifs);
    }

    getGifs();

    return (
        <div>
            <h3>{category}</h3>
        </div>
    )
}
