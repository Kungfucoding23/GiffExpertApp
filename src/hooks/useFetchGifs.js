import {useState, useEffect} from 'react';
import { getGifs } from '../helpers/getGifs';
//este es un custom hook, funciona como un functional component 

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });
    useEffect(() => {
        getGifs(category)
            .then(imgs => {                
                    setState({
                        data: imgs,
                        loading: false
                    }); 
            })
    }, [category])
    return state; // {devuelve el objeto}
}
