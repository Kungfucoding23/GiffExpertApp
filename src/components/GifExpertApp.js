import React, {useState} from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = (e) => {
    //     // 'HunterXHunter'
    //     setCategories( category => [...category, 'HunterXHunter']);
    // }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}/>
            <hr />
            {/* <button onClick={handleAdd}>Agregar</button> */}

            <ol>
                {
                    categories.map(category => (
                        <GifGrid 
                        key={category}
                        category={category}/>
                    ))
                }
            </ol>            
        </>
    )
}

export default GifExpertApp;