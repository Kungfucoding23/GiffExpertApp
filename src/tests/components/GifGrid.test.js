import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import {shallow} from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../components/GifGrid');


describe('pruebas en <GifGrid />', () => {

    const category = 'One Punch';

    test('debe mostrarse ok', () => {
        // const gifs = [{
        //     id: 'ABC',
        //     url: 'https://localhost/cualquier/cosa.jpg',
        //     title: 'Cualquier cosa'
        // }];
        // useFetchGifs.mockReturnValue({
        //     data: gifs,
        //     loading: true
        // });
        const wrapper = shallow(<GifGrid category={category}/> );
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/> );
        expect(wrapper).toMatchSnapshot();

    })
    
    
})
