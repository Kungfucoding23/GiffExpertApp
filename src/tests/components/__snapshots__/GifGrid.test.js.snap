// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid /> debe mostrarse ok 1`] = `""`;
